File: tests/fixtures/sample-docs.go
package main

// UserService handles user management operations.
type UserService struct {
	db *Database
}

// CreateUser creates a new user with the given name.
func (s *UserService) CreateUser(name string) (*User, error) {
	return nil, nil
}

// DeleteUser removes a user by ID. Returns an error if not found.
func (s *UserService) DeleteUser(id int) error {
	return nil
}

func (s *UserService) noDocsMethod() {}

// MaxRetries is the maximum number of retry attempts.
const MaxRetries = 3

// DefaultTimeout defines the default request timeout in seconds.
var DefaultTimeout = 30

func helperNoDoc() {}

File: tests/fixtures/sample-docs.rs
/// A service for managing user accounts.
pub struct UserService {
    db: Database,
}

/// Create a new user with the given name.
///
/// # Arguments
/// * `name` - The user's display name
pub fn create_user(name: &str) -> Result<User, Error> {
    todo!()
}

/// Delete a user by ID. Returns an error if not found.
pub fn delete_user(id: u64) -> Result<(), Error> {
    todo!()
}

fn no_docs_helper() {}

/// Maximum number of retry attempts.
pub const MAX_RETRIES: u32 = 3;
